src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/Shared/CohostDevToolsEndpointTest.cs

```diff
@@ -0,0 +1,133 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.CodeAnalysis.Razor.Protocol.DevTools;
+using Microsoft.CodeAnalysis.Testing;
+using Roslyn.LanguageServer.Protocol;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostDevToolsEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Fact]
+    public async Task GeneratedDocumentContents_CSharp()
+    {
+        var input = """
+                @{
+                    var message = "Hello World";
+                }
+                <div>@message</div>
+                """;
+
+        await VerifyGeneratedDocumentContentsAsync(input, GeneratedDocumentKind.CSharp, "message", ".g.cs");
+    }
+
+    [Fact]
+    public async Task GeneratedDocumentContents_Html()
+    {
+        var input = """
+                @{
+                    var message = "Hello World";
+                }
+                <div>@message</div>
+                """;
+
+        await VerifyGeneratedDocumentContentsAsync(input, GeneratedDocumentKind.Html, "div", ".g.html");
+    }
+
+    [Fact]
+    public async Task GeneratedDocumentContents_Formatting()
+    {
+        var input = """
+                @{
+                    var message = "Hello World";
+                }
+                <div>@message</div>
+                """;
+
+        await VerifyGeneratedDocumentContentsAsync(input, GeneratedDocumentKind.Formatting, "message", ".formatting.cs");
+    }
+
+    [Fact]
+    public async Task GetTagHelpers_ReturnsJson()
+    {
+        var input = """
+                @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+                <div>Test content</div>
+                """;
+
+        var razorDocument = await CreateDocumentAsync(input);
+        var endpoint = new CohostTagHelpersEndpoint(IncompatibleProjectService, RemoteServiceInvoker);
+        
+        var request = new TagHelpersRequest
+        {
+            TextDocument = new TextDocumentIdentifier { DocumentUri = razorDocument.CreateDocumentUri() },
```

Comment:
In order to call CreateDocumentUri, this file needs `using Microsoft.CodeAnalysis.ExternalAccess.Razor;` at the top